### ISSUE TITLE
Getting 401 on /identity example

### ISSUE THREAD

**Issue by SamPotts** [bartve/disconnect#21]
Cheers for the lib. Well handy. I'm just having a little trouble... 

I'm passing back the access data (stored in session using session-cookie) using:

```javascript
app.use('/identity', function(req, res) {
    var dis = new Discogs(req.session.accessData);

    dis.getIdentity(function(err, data) {
    	console.log(err);
        res.end(data);
    });
});
```

and `req.session.accessData` has all the right keys present but I still get this back:

> AuthError: You must authenticate to access this resource.

Any help would be appreciated 👍

**Comment by bartve** [comment]
Does it work with Discogs Auth as the authentication method?

**Comment by SamPotts** [comment]
Apologies for my lack of knowledge but how would I do that?

**Comment by bartve** [comment]
Like this: https://www.discogs.com/developers/#page:authentication,header:authentication-discogs-auth-flow

**Comment by SamPotts** [comment]
So does the identity example no longer work then? I was just following the readme. Is it out of date?

**Comment by SamPotts** [comment]
I think this was closed by mistake? It's still not working. Could you maybe try the examples? The identity example does not work.

**Comment by bartve** [comment]
Oops, I didn't mean to close this indeed! Probably clicked the wrong button. Sorry! I will probably have time to look into this tomorrow.

**Comment by SamPotts** [comment]
Cheers! I'm trying to help a mate out by building him an app and stuck at the moment :-(

**Comment by bartve** [comment]
I did some tests and I don't seem to be able to reproduce the issue. Are you sure the access token is correctly persisted and passed? The OAuth flow is quite complicated and you must pass the final access token and not a token from an earlier step. Both Discogs Auth and OAuth authentication in combination with the identity endpoint seem to be working fine here. I also would have expected others to complain if this was broken.

**Comment by SamPotts** [comment]
Thanks for taking a look. I'll send some more info when I get home.

**Comment by SamPotts** [comment]
I'm using cookie-session to persist the `accessData` and `requestData`: 

```javascript
cookieSession = require('cookie-session');

app.use(cookieSession({
	keys: ['key1', 'key2']
}));
```

and then I store what's returned in `accessData` as below in the callback after oauth:

```javascript
// Persist "accessData" here for following OAuth calls 
req.session.accessData = accessData;
```

`console.log(accessData === req.session.accessData)` straight after returns `true` as does `console.log(dis.authenticated())`. `accessData` returns the following (some parts hidden): 

```javascript
{ 
  method: 'oauth',
  level: 0,
  consumerKey: '...',
  consumerSecret: '...',
  token: '...',
  tokenSecret: '...',
  authorizeUrl: 'https://www.discogs.com/oauth/authorize?oauth_token=...' 
}
```

In the method that uses getIdentity a `console.log(req.session.accessData)` returns the same as above but it still doesn't seem to authenticate. I tried passing the `token` and `tokenSecret` properties rather than the whole object but it didn't seem to work either... 

Which seems to be very similar to the `requestData` used in the earlier step? Sorry if it's a little confusing. I can send you the `server.js` file if it helps.

**Comment by SamPotts** [comment]
OK, I figured it out. It was earlier in the process. Getting the access token was actually failing silently (as I wasn't checking the `err` argument value) due to the `oauth_verifier` query parameter being missing. It was because I wasn't accessing the `query` property when parsing the URL:

```javascript
var query = url.parse(req.url, true).query;
```

Sorry about all this. Thanks for looking into it for me though 👍

**Comment by bartve** [comment]
No problem. Glad you got it to work :+1:

**Comment by codecreate** [comment]
Hi. I just found this exampe of a similar problem I'm having. I'm using the oAuth method and it's working fine up until trying the /identity test. The flow completes succesfully bringing me back to my app succesfully but I get a 401 auth error on /identity and when I investigate I discover that the accessData that I persisted in the /callback is the same as the requestData that I persisted in the /authorise. I'm persisting both those objects in app.locals. Any idea why they are the same? Neither the /authorise or /callback return any err.

Thanks
